pkg/probe/bgp_neighbor_routes_test.go

```diff
@@ -0,0 +1,57 @@
+package probe
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/testutil"
+)
+
+func TestBGPNeighborPathsIPv4(t *testing.T) {
+	c := newFakeClient()
+	c.prepare("api/v2/monitor/router/bgp/paths", "testdata/router-bgp-paths-v4.jsonnet")
+	r := prometheus.NewPedanticRegistry()
+	if !testProbe(probeBGPNeighborPathsIPv4, c, r) {
+		t.Errorf("probeBGPNeighborPathsIPv4() returned non-success")
+	}
+
+	em := `
+    # HELP fortigate_bgp_neighbor_best_paths_ipv4_count Count of best paths for an BGP neighbor
+    # TYPE fortigate_bgp_neighbor_best_paths_ipv4_count gauge
+    fortigate_bgp_neighbor_best_paths_ipv4_count{neighbor_ip="10.0.0.1",vdom="root"} 1
```

Comment:
Maybe rename this more in line of `fortigate_bgp_neighbor_propagated_paths_ipvX` and `fortigate_bgp_neighbor_propagated_best_paths_ipvX`? @bl0m1 @bluecmd

Reply:
I'm open for this, The current naming is just a copy paste of the API endpoint and value naming.

Reply:
All paths are `propagated`, the `best_paths` are just the ones that are active / in-use. Since the algorithm for selecting these are called "BGP Best Path Selection" I think it makes sense to stick to the terminology.